source/docs/flow_control.md

```diff
@@ -278,4 +278,59 @@ called on creation.
 
 ### HTTP implementation details
 
-TODO(alyssawilk) implement and document.
+HTTP flow control is extremely similar to HTTP/2 flow control, with the main exception being that
+the method used to halt the flow of downstream/upstream data is to disable reads on the underlying
+Network::Connection.  As the TCP data stops being consumed the peer will eventually fill their
+congestion window and stop sending.
+
+As with HTTP/2, a given stream may end in a state where the connection has had
+`readDisable(true)` called.   When a new stream is created on that connection
+for a subsequent request, any outstanding `readDisable(true)` calls are unwound
+in `Http::Http1::ConnectionImpl::newStream()`.
+
+Filter and network backups are identical in the HTTP and HTTP/2 cases and are
+documented above.  Codec backup is slightly different and is documented below.
+
+# HTTP codec downstream send buffer
+
+As with the HTTP/2 codec recv buffer, the HTTP codec send buffer is only expected to have data pass
+through it.  It should never back up.  Still, in the event it does, the high
```

Comment:
I've been thinking about this more, and I question whether we should code for cases that should never happen. I feel that same way about the HTTP/2 recv buffer. That should never back up either but we have code for it. WDYT about potentially keeping the watermark buffers in place but just asserting that the callbacks never get called vs. having all this extra code? (In both cases assuming we have test coverage it feels like the tests must be extremely synthetic since these cases should never happen?)